src/providers/ThemeProvider.tsx

```diff
@@ -51,13 +64,77 @@ export const ThemeProvider: React.FC<
         path: '/',
         sameSite: true,
       });
-      setTheme(themeMode === 'dark' ? 'dark' : 'light');
+      setThemeModeState(themeMode === 'dark' ? 'dark' : 'light');
     }
   }, [themeMode, isDarkMode, setCookie]);
 
-  const activeTheme = theme === 'dark' ? darkTheme : lightTheme;
+  const activeTheme = useMemo(() => {
+    let currentTheme = themeModeState === 'dark' ? darkTheme : lightTheme;
+
+    if (
+      partnerCustomizedTheme ||
+      (cookie.partnerThemeUid !== undefined &&
+        cookie.partnerThemeUid !== 'undefined')
+    ) {
+      // Merge partner themeModeState attributes into the base themeModeState
+      const mergedTheme = deepmerge(currentTheme, {
+        typography: {
+          fontFamily:
+            updatedTheme?.typography &&
+            currentTheme.typography.fontFamily &&
+            currentCustomizedTheme?.typography.includes('Sora')
+              ? sora.style.fontFamily.concat(currentTheme.typography.fontFamily)
+              : currentTheme.typography.fontFamily,
+        },
+        palette: {
+          primary: {
+            main: updatedTheme?.palette?.primary
```

Comment:
why is the interface different than classic theme? Please keep it align to avoid any issues